api/routes_ipns.go

```diff
@@ -113,6 +113,18 @@ func (api *API) publishToIPNSDetails(c *gin.Context) {
 	Respond(c, http.StatusOK, gin.H{"response": "ipns entry creation sent to backend"})
 }
 
+// getIPNSRecordsPublishedByUser is used to fetch IPNS records published by a user
+func (api *API) getIPNSRecordsPublishedByUser(c *gin.Context) {
+	username := GetAuthenticatedUserFromContext(c)
+	ipnsManager := models.NewIPNSManager(api.dbm.DB)
```

Comment:
similarly to `UserManager`, does this have to be created on a per-API-call basis? I think it can persist through the lifespan of `API`

Reply:
Agreed, perhaps all the model managers should persist through the lifespan of API?